cachito/web/api_v1.py

```diff
@@ -716,3 +718,87 @@ def get_content_manifest_by_requests():
     if len(requests) > 1:
         deep_sort_icm(assembled_icm)
     return send_content_manifest_back(assembled_icm)
+
+
+class RequestMetricsArgs(pydantic.BaseModel):
+    """Query parameters for /request-metrics endpoint."""
+
+    start_date: Optional[date] = None
+    end_date: Optional[date] = None
+
+
+@api_v1.route("/request-metrics", methods=["GET"])
+def get_request_metrics():
+    """Return a list of completed requests with a final state information."""
+    max_per_page = flask.current_app.config["CACHITO_MAX_PER_PAGE"]
+    args = RequestMetricsArgs(**flask.request.args)
+    query = RequestState.get_final_states_query().order_by(RequestState.request_id.desc())
+
+    if args.start_date:
+        query = query.filter(RequestState.updated >= args.start_date)
+    if args.end_date:
+        query = query.filter(RequestState.updated < args.end_date + timedelta(days=1))
+
+    pagination_query = query.paginate(max_per_page=max_per_page)
+    return flask.jsonify(
+        {
+            "items": [
+                {
+                    "id": state.request_id,
+                    "final_state": RequestStateMapping(state.state).name,
+                    "final_state_reason": state.state_reason,
+                    "finished": state.updated.isoformat(),
+                    "duration": state.duration,
```

Comment:
And maybe `initial_inactive` would be useful too (time between state 1 and state 2)

Reply:
As we discussed previously, the difference of time between two states would not have the same meaning.
In Cachito activity is represented by async tasks. And sometimes we set state on the start of that task, sometime in the end. Also we have couple of cases where it is set in the middle of execution.
We can not calculate the actual inactivity time as well. Inactivity in our case could mean difference between one tasks end and the following task start. And we have no such data.

However we are preparing changes that could help to fill the gap. We are going to introduce a table which will track duration of each task. Base on the data we could extend the request metrics endpoint. But currently implementing any of such metrics could just mislead users.

Reply:
I would not build logic to adjust the wrong assumption.
For the certain case, you provided, it would better if Cachito could respond explicitly about a request status to avoid such guessing.

Reply:
To a non-human client, the only indicator of request activity is the state updates. This metric is trying to capture apparent inactivity. But if you don't see a use for it you can drop it, just make sure to reflect that in the planned SLI work.

Reply:
I was thinking it might be a good idea to change the first state to something like `queued` or simply `created`. This could make it more clear, not sure if there'll be any impact though.

I do think that a `time_in_queue` metric would be very useful, I'd like to keep an eye on that. I believe that could be done in a separate PR, if @serg-cymbaluk prefers.